src/command/which.rs

```diff
@@ -34,20 +63,29 @@ impl Command for Which {
             None => System::path()?,
         };
 
-        match which_in(&self.binary, Some(path), cwd) {
+        let path = match tool_path {
+            Some(tool_path) => {
+                default_path.push(tool_path.as_os_str());
```

Comment:
If this is still needed, it's probably best to use the `tool_path` as the whole `path` for this case. We can't just `push` an extra path onto the end of the `PATH` string, because we need to take into account the separator character, which is platform-specific. Inside `System::path` and `Image::path`, we use the `envoy` crate to handle that for us, but in this case I think we'll know definitively that the binary will be found at `tool_path`, so we don't need to include any extra paths when calling `which_in`.